### PR TITLE
Adding relative method in path interface

### PR DESCRIPTION
**Pull request by jpuri**
Hey @kennetpostigo , great project.
I was checking it out. I want to contribute to the project and thus I created this PR for a method.

I want to implement `path.relative` method. What I understand about the method is:

1. returns the relative path from -> to
2. if from and to each resolve to the same path return zero length string
3. if either of from or to are zero length string use current working directory instead of them

Please suggest if I am missing out on anything.

### CODE REVIEW

**Comment by kennetpostigo** [comment]
Hey @jpuri ! that sounds just about right. My strategy so far in implementing `path` operators is to check out the [node.js doc examples](https://nodejs.org/dist/latest-v9.x/docs/api/path.html#path_path_relative_from_to) add a test case with that example. I then also go to node.js and I then try out a bunch of examples that I want to use as test cases and copy the result from node.js repl, as the expected output for the tests. 

I would add test cases with:
- absolute paths
- relative paths
- unnormalized paths
- normalized paths

**Comment by jpuri** [comment]
I had some troubles in running tests with jbuilder with latest version of ocaml `4.06.0` but it works perfect in version `4.02.3`.

**Comment by kennetpostigo** [comment]
Interesting, do you think you can describe the issues @jpuri ?

**Comment by jpuri** [comment]
Seems to be trouble in some opam packages, its not able to install all required packages for running the test cases:

```
Jyotis-Air:reason-node jyotipuri$ opam install alcotest
The following actions will be performed:
  ∗  install topkg    0.9.1                   [required by cmdliner, astring, fmt]
  ∗  install cmdliner 1.0.2                   [required by alcotest]
  ∗  install astring  0.8.3                   [required by alcotest]
  ∗  install fmt      0.8.5                   [required by alcotest]
  ∗  install alcotest 0.8.1
===== ∗  5 =====
Do you want to continue ? [Y/n] y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[alcotest] Archive in cache
[astring] Archive in cache
[cmdliner] Archive in cache
[fmt] Archive in cache
[topkg] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[ERROR] The compilation of topkg failed at "ocaml pkg/pkg.ml build --pkg-name topkg --dev-pkg false".

#=== ERROR while installing topkg.0.9.1 =======================================#
# opam-version 1.2.2
# os           darwin
# command      ocaml pkg/pkg.ml build --pkg-name topkg --dev-pkg false
# path         /Users/jyotipuri/.opam/4.06.0/build/topkg.0.9.1
# compiler     4.06.0
# exit-code    2
# env-file     /Users/jyotipuri/.opam/4.06.0/build/topkg.0.9.1/topkg-49980-c61e7b.env
# stdout-file  /Users/jyotipuri/.opam/4.06.0/build/topkg.0.9.1/topkg-49980-c61e7b.out
# stderr-file  /Users/jyotipuri/.opam/4.06.0/build/topkg.0.9.1/topkg-49980-c61e7b.err
### stdout ###
# Cannot find file topfind.
### stderr ###
# Unknown directive `require'.



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
The following actions were aborted
  ∗  install alcotest 0.8.1
  ∗  install astring  0.8.3
  ∗  install cmdliner 1.0.2
  ∗  install fmt      0.8.5
The following actions failed
  ∗  install topkg 0.9.1
No changes have been performed
```

But I think its not specific to project `reason-node`.

**Comment by kennetpostigo** [comment]
@jpuri you shouldn't have to use opam, although you could but I don't recommend it. This is probably my fault for not documenting the commands I use to build, test, etc.

So this is usually how the flow of things would go:

1. esy install
2. esy build
3. esy make test

When using esy, you have to prefix commands with esy in order for it to pick up the local environment/sandbox.

**Comment by jpuri** [comment]
Yep I did not prefixed easy before `make test`. That seems to have created all this trouble.

**Comment by kennetpostigo** [comment]
It works now?

**Comment by jpuri** [comment]
yep works well now

**Comment by jpuri** [comment]
Still WIP. Going slow as I do not have a lot of time.

**Comment by kennetpostigo** [comment]
@jpuri no worries :) Thank you so much for putting time into it!

**Comment by jpuri** [comment]
Hey @kennetpostigo , can you plz check the changes and share ur feedbacks.

**Comment by kennetpostigo** [comment]
@jpuri yes, definitely! I'll review it today

**Comment by andreypopp** [comment]
Hi! I suspect that macOS build fails because esy doesn't produce any output during build process when tty isn't available (like when run on CI). Sometimes build process can take longer than 10m (I saw this earlier versions of esy too), this can be because travis is out of capacity for macOS machines, for example. See https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received for a workaround.

Next release (soon) I'll fix esy output to provide some progress indication even if no tty is available.

**Comment by kennetpostigo** [comment]
@andreypopp thank you!

**Comment by kennetpostigo** [comment]
Hey @jpuri , The project has been renamed to `lwt-node` so there have been quite a few changes. Sorry for the trouble, do you think you can sync this PR back up with master?

**Comment by jpuri** [comment]
Hey @kennetpostigo , sure I will sync up the PR and work n feedbacks you shared. Thx.

**Comment by jpuri** [comment]
Hey @kennetpostigo , point-1 above is not clear to me: `output after diffing both paths should be relative to the current working directory`. I also checked and compared with NodeJS output and you are right its different.

I have implemented 2, 3 and I am working on improving test coverage. More tests are required - PR is not complete. I requested a review so that I would know that my approach is correct.

**Comment by kennetpostigo** [comment]
@jpuri after messing around in the node repl, the original thought was wrong, the reason why the output is different is because in node.js it expects the input to be directories and not files, so if you change:
```js
path.relative("bar/index.html", "index.html")
```

to

```js
Path.relative("bar", "index.html")
```

you will get "../index.html".

Personally I prefer the manner in which it is implemented now, it feels more precise to me. However, if porting code directly from node.js to lwt-node this might be one of those "gotchas". I leave the decision up to you of whether you want to leave it as is, or change it to match node.js.

**Comment by jpuri** [comment]
That was my thought also that nodejs is expecting directory and not file.
The current implementation as I have done will break if user passed not file-path but directly. I would prefer to keep this function same as node's. But it may be useful to have another like `relativeFilePath`.

**Comment by kennetpostigo** [comment]
@jpuri thats a good idea! :)

**Comment by jpuri** [comment]
`Path.resolve([""])` should return current working directory. This is not the case atm, is this bug or intended.

**Comment by kennetpostigo** [comment]
@jpuri good catch!

The quick fix could be:
```reason
let resolve = paths => {
  let keepJoining = ref(true);
  let rePath = ref("");
  let onPath = (path, index, e) =>
    e ?
      {
        keepJoining := false;
        rePath := path ++ sep ++ rePath^;
      } :
      (
        switch path {
        | ""
        | "./"
        | ".\\" => Sys.getcwd() /* <----- fix*/
        ...
```

**Comment by jpuri** [comment]
PR looks complete to me now, also I added change and test coverage for resolve method.

**Comment by kennetpostigo** [comment]
@jpuri thank you! I'll take a look at this ASAP :)

File: src/renodeUtils.re [hunk 1]
```diff
@@ -6,6 +6,47 @@ let charSep =
   | _ => '/'
   };
 
+let rec removeCommonPath = (~path1, ~path2) => {
```
Comment (jpuri) [inline src/renodeUtils.re]: These switches can be replaced by ternary operators may be, but they are best at handling exceptions. Plz share if you have better suggestions about code here.

Comment (kennetpostigo) [reply src/renodeUtils.re]: I personally prefer switches, they make refactoring easier and more explicit the exceptions that are being handled. These look great!

**Comment by kennetpostigo** [comment]
The changes look great! Thank you for your work! @jpuri

**Comment by jpuri** [comment]
Hi @kennetpostigo, I have squashed the changes, can you plz merge as I do not have that option enabled.

**Pull request merged**